### DISCUSSION TITLE
Remove ambient light noise from PPG recordings

### DISCUSSION THREAD

**Discussion by JackCaster** [neuropsychology/NeuroKit#845] [Q&A] [answered]
I have collected PPG data using a 3-channel PPG sensor. This sensor read PPG values in the red, green, and IR spectrum. Additionally, it provides a measure of the ambient light noise. What is the recommended way of removing the noise component from the PPG signal? 

My first attempt was to remove the noise equally from the 3 PPG channels, but it does not seem the ideal approach---I expect the noise contribution to the different channels to be different. Would you be able to help?

**Comment by DominiqueMakowski** [comment]
Mmh I have no experience with this type of PPG sensor. Could you attach a plot of the 3 channels just to see what they look like?

**Reply by JackCaster** [reply]
Sure thing!
![ppg-index](https://github.com/neuropsychology/NeuroKit/assets/18400141/8c243508-da00-4e28-9afe-7ed900bb918f)

This is collected on the fingertip using a [Polar Verity Sense](https://www.polar.com/en/products/accessories/polar-verity-sense)

**Reply by DominiqueMakowski** [reply, marked as answer]
It seems like there won't be much difference in terms of peak detection: the peaks are clear in all channels. I'd just peak maybe say the orange one (or the one in which peaks have the most amplitude), and treat as your main PPG signal. Or maybe average them, but I'm not sure there would be lots of benefits to that 🤷

**Reply by JackCaster** [reply]
Right, the peaks are quite good at this location, but we are also experimenting in areas where the signal is a bit weaker. On the forehead it is still quite good, but that is not always the case.
![ppg-forehead](https://github.com/neuropsychology/NeuroKit/assets/18400141/8e5fd65a-50cd-48fc-8d05-b8e5f3c95d01)

As you also suggested, we are averaging the three channels, mean-center the signal, and removing the mean-centered ambient noise as well.
